packages/loader/container-loader/src/container.ts

```diff
@@ -1137,6 +1138,11 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             throw new Error("Not attached");
         }
         const storageService = await this.service.connectToStorage();
+
+        // Enable prefetching for the service unless it has a caching policy set otherwise:
+        if (this.service.policies?.caching === ServiceCachingPolicy.NoCaching) {
+            return storageService;
+        }
         return new PrefetchDocumentStorageService(storageService);
```

Comment:
I think you should create the prefetchDocumentstorageserice and then if ploicy if of no prefetching then call prefetchstorage.stopPrefetch() which disables prefetching. The benefit is that then all storage operations will go through prefetchstorageservice and if any operation is required on strorage read and write we don't need to spill it all around the repo, we can just do that in prefetch storage service. @vladsud